File: leetcode/tree/invert_binary_tree.py
from common import TreeNode
from typing import List

class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        def helper(root):
            if not root:
                return
            
            root.left, root.right = root.right, root.left
            helper(root.left)
            helper(root.right)
        helper(root)
        return root

if __name__ == "__main__":
    root = TreeNode(1)
    root.left = TreeNode(2)
    root.right = TreeNode(3)
    root = Solution().invertTree(root)
    print(root.val, root.left.val, root.right.val)

File: leetcode/tree/invert_binary_tree.go
package main

import "fmt"

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func invertTree(root *TreeNode) *TreeNode {
	helper(root)
	return root
}

func helper(root *TreeNode) {
	if root == nil {
		return
	}
	root.Left, root.Right = root.Right, root.Left
	helper(root.Left)
	helper(root.Right)
}

func main() {
	root := &TreeNode{3, nil, nil}
	root.Left = &TreeNode{9, nil, nil}
	root.Right = &TreeNode{20, nil, nil}
	root.Right.Left = &TreeNode{15, nil, nil}
	root.Right.Right = &TreeNode{7, nil, nil}
	out := invertTree(root)
	fmt.Println(out.Val, root.Left.Val, root.Right.Val)
}
